crates/core/src/repository.rs

```diff
@@ -416,8 +416,28 @@ impl<P, S> Repository<P, S> {
     /// [`RepositoryErrorKind::ListingRepositoryConfigFileFailed`]: crate::error::RepositoryErrorKind::ListingRepositoryConfigFileFailed
     /// [`RepositoryErrorKind::MoreThanOneRepositoryConfig`]: crate::error::RepositoryErrorKind::MoreThanOneRepositoryConfig
     pub fn config_id(&self) -> RusticResult<Option<ConfigId>> {
-        let config_ids = self
-            .be
+        return self.config_id_with_backend(&self.be);
+    }
+
+    /// Returns the Id of the config file
+    ///
+    /// # Errors
+    ///
+    /// * [`RepositoryErrorKind::ListingRepositoryConfigFileFailed`] - If listing the repository config file failed
+    /// * [`RepositoryErrorKind::MoreThanOneRepositoryConfig`] - If there is more than one repository config file
+    ///
+    /// # Arguments
+    ///
+    /// * `be` - The backend to use
+    ///
+    /// # Returns
+    ///
+    /// The id of the config file or `None` if no config file is found
+    ///
+    /// [`RepositoryErrorKind::ListingRepositoryConfigFileFailed`]: crate::error::RepositoryErrorKind::ListingRepositoryConfigFileFailed
+    /// [`RepositoryErrorKind::MoreThanOneRepositoryConfig`]: crate::error::RepositoryErrorKind::MoreThanOneRepositoryConfig
+    fn config_id_with_backend(&self, be: &dyn WriteBackend) -> RusticResult<Option<ConfigId>> {
```

Comment:
Here `self` is used only to craft the error.
I am not sure this is a reason to change anything. Probably it is not a reason to change.
I am just wondering if:
- the method signature could be simplified
- the method could be implemtend for `WriteBackend` trait

Reply:
yeah, I was just trying to find something to say about the changes.
Please don't act on my comment ^^